mypostcard/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bytes"
+	"embed"
+	"flag"
+	"fmt"
+	"image"
+	"image/draw"
+	"image/png"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/freetype/truetype"
+	"golang.org/x/image/font"
+	"golang.org/x/image/math/fixed"
+)
+
+//go:embed LICENSE
+var license string
+
+//go:embed postcard.png
+var postcardPNG []byte
+
+//go:embed stamp*.png
+var stampsPNG embed.FS
+
+//go:embed ClarendonBT.ttf
+var clarendonBT []byte
+
+//go:embed *.html *.css LICENSE
+var webUI embed.FS
+
+func main() {
+	listenAddr := flag.String("l", ":8000", "Listen address")
+	printLicense := flag.Bool("license", false, "Print license")
+	flag.Parse()
+	if *printLicense {
+		fmt.Println(license)
+		return
+	}
+	rand.Seed(time.Now().UnixNano())
+	http.Handle("/", http.FileServer(http.FS(webUI)))
+	http.HandleFunc("/postcard", postcardHandler)
+	fmt.Println("Listen and serve on", *listenAddr)
+	err := http.ListenAndServe(*listenAddr, nil)
+	if err != nil {
+		fmt.Println("ERROR", err)
+	}
+}
+
+func postcardHandler(w http.ResponseWriter, r *http.Request) {
+	text := r.FormValue("postcard")
+	text = strings.ReplaceAll(text, "\r", "")
+	lines := strings.Split(text, "\n")
+	postcard := generatePostcard(lines)
+	var b bytes.Buffer
+	err := png.Encode(&b, postcard)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "image/png")
+	w.Header().Set("Content-Length", strconv.Itoa(len(b.Bytes())))
+	_, err = w.Write(b.Bytes())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func generatePostcard(lines []string) image.Image {
```

Comment:
Prettify this function, add comments and maybe break it up a little.

Reply:
There is no golangci-lint for go1.16 yet, but I used `vec` and format using my editor.
I'm keeping the function as a single block for the blog.
If you think there is something specific inside the function that doesn't clear I'll add more information in the function description or line comment.